### PR TITLE
when there is only one project matching the mainclass, projectName is not required for evaluation.

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/Constants.java [hunk 1]
```diff
@@ -14,4 +14,5 @@
 public final class Constants {
     public static final String PROJECTNAME = "projectName";
     public static final String DEBUGGEE_ENCODING = "debuggeeEncoding";
+    public static final String MAINCLASS = "mainClass";
```
Comment (yaohaizh) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/Constants.java]:
```java
MAIN_CLASS
```

Comment (andxu) [reply com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/Constants.java]: Done  with the same issue fixed for project name

File: com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/eval/JdtEvaluationProvider.java [hunk 2]
```diff
@@ -119,6 +132,91 @@ public void initialize(IDebugAdapterContext context, Map<String, Object> props)
         }
     }
 
+    /**
+     * Prepare a list of possible java projects in workspace which contains the main class.
+     *
+     * @param mainclass the main class specified by launch.json for finding possible projects
+     */
+    private void initializePossibleProjects(String mainclass) {
```
Comment (yaohaizh) [inline com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/eval/JdtEvaluationProvider.java]: initializeProjectCandidates

File: com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/eval/JdtEvaluationProvider.java [hunk 3]
```diff
@@ -119,6 +132,91 @@ public void initialize(IDebugAdapterContext context, Map<String, Object> props)
...
+                possibleProjects = validProjects;
+                logger.severe("Multiple projects are valid for evaluation.");
+                throw new IllegalStateException("Multiple projects are found, please specify projectName in launch.json.");
+            }
+
+        } catch (Exception ex) {
+            // ignore
+        }
+
+        logger.severe("Cannot evaluate when project is not specified.");
```
Comment (yaohaizh) [inline com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/eval/JdtEvaluationProvider.java]: "Cannot evaluate when `the` project is not specified

**Pull request merged**